File: app/src/main/java/com/github/kalsmic/android/pomodorotimer/BaseActivity.java
package com.github.kalsmic.android.pomodorotimer;

import android.content.Intent;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

public class BaseActivity extends AppCompatActivity {

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater menuInflater = getMenuInflater();
        menuInflater.inflate(R.menu.options_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        Intent goToMenuOption = new Intent();
        switch (item.getItemId()) {

            case R.id.menu_title_sounds:
                goToMenuOption.setClass(this, NotificationsActivity.class);
                startActivity(goToMenuOption);
                return true;

            case R.id.menuAbout:
                goToMenuOption.setClass(this, AboutActivity.class);
                startActivity(goToMenuOption);
                return true;

            case R.id.menuShare:
                Toast.makeText(this, "Menu Share Clicked", Toast.LENGTH_SHORT).show();
                return true;

            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/com/github/kalsmic/android/pomodorotimer/BaseActivity.kt
package com.github.kalsmic.android.pomodorotimer

import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity

open class BaseActivity : AppCompatActivity() {
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val menuInflater = menuInflater
        menuInflater.inflate(R.menu.options_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val goToMenuOption = Intent()
        when (item.itemId) {
            R.id.menu_title_sounds -> {
                goToMenuOption.setClass(this, NotificationsActivity::class.java)
                startActivity(goToMenuOption)
                return true
            }

            R.id.menuAbout -> {
                goToMenuOption.setClass(this, AboutActivity::class.java)
                startActivity(goToMenuOption)
                return true
            }

            R.id.menuShare -> {
                goToMenuOption.setClass(this, ShareActivity::class.java)
                startActivity(goToMenuOption)
                return super.onOptionsItemSelected(item)
            }

            else -> return super.onOptionsItemSelected(item)
        }
    }
}